src/feature/Registration/index.jsx

```diff
@@ -1,22 +1,63 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
+import { useDispatch } from "react-redux";
+import { registrationSuccess } from "../../redux/actions/auth-actions";
 import styles from "./Registration.module.css";
 import { Link, useNavigate } from "react-router-dom";
 import logo from "../../images/sniper.png";
 import Button from "../../components/Button";
 import Input from "../../components/Input";
-import { validateEmail, validatePassword } from "../../utils/validation";
+import {
+  validateEmail,
+  validatePassword,
+  validateConfirmPassword,
+} from "../../utils/validation";
+import Cookies from "universal-cookie";
 
 const Registration = () => {
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
   const [confirmPassword, setConfirmPassword] = useState("");
```

Comment:
Refactor this into a single state:

```suggestion
  const signupForm = useState({
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    confirmPassword: '',
  });
```

For each input element, give it the matching name key. For example:
```
<input name="firstName" onChange={setFormValue} />
```

Then to update form value, simple do:

```js
const setFormValue = (el) => {
    setSignupForm(form => ({
      ...form,
      [el.target.name]: el.target.value
    }))
  }
```

After these, clean up the code to remove all the repetative handleChange function.